### PR TITLE
Support Provider Network in the KubeVirt CloudSpec in the Seed

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
This PR adds supports for provider networks for KubeVirt. Using this would allow our users to support listing Subnets and VPCs from the seed without the need to grant special permissions on the infra kubeconfig. 

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Support Provider Network VPCs and Subnets in the Seed Object
```

**Documentation**:

```documentation
None
```

### CODE REVIEW

File: pkg/apis/kubermatic/v1/datacenter.go [hunk 1]
```diff
@@ -829,6 +829,27 @@ type DatacenterSpecKubevirt struct {
...
+type ProviderNetwork struct {
+	Name string `json:"name"`
+	VPC  []VPC  `json:"vpc"`
+}
+
+// VPC  is a virtual network dedicated to a single tenant within a KubeVirt, where the resources in the VPC
+// is isolated from any other resources within the KubeVirt infra cluster.
+type VPC struct {
+	Name   string   `json:"name"`
+	Subnet []Subnet `json:"subnet,omitempty"`
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/datacenter.go]: Should this be `Subnets` (plural)? Same with the VPC field in ProviderNetwork.

File: pkg/apis/kubermatic/v1/datacenter.go [hunk 2]
```diff
@@ -843,7 +864,7 @@ type KubeVirtInfraStorageClass struct {
 	Name string `json:"name"`
 	// Optional: IsDefaultClass. If true, the created StorageClass in the tenant cluster will be annotated with:
 	// storageclass.kubernetes.io/is-default-class : true
-	// If missing or false, annotation will be:
+	// If missing or false,  annotation will be:
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/datacenter.go]: Did you mean to insert a "the" here to fix the grammar?

Comment (moadqassem) [reply pkg/apis/kubermatic/v1/datacenter.go]: Interesting, I don't see this change in my revision 🤔

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**